### PR TITLE
refactor: replace nano contract state call for history call

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- Replace call to nano contract state by other call to get nano contract information

This is important because the "/state" endpoint is not consistent as it can return "not found" while the new nano contract was not confirmed by a block.

Closes: https://github.com/HathorNetwork/hathor-wallet-mobile/issues/510

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/sagas/nanoContract.js [hunk 1]
```diff
@@ -111,9 +111,9 @@ export function* registerNanoContract({ payload }) {
     return;
   }
 
-  let ncState = null;
+  let ncHistory = null;
   try {
-    ncState = yield call([ncApi, ncApi.getNanoContractState], ncId);
+    ncHistory = yield call([ncApi, ncApi.getNanoContractHistory], ncId, 1);
```
Comment (pedroferreira1) [inline src/sagas/nanoContract.js:116]: I understand that you are using this to check if the nano contract exists but semantically I prefer doing this like the desktop wallet.

```
  try {
    const response = yield call([wallet, wallet.getFullTxById], ncId);
    tx = response.tx;
  } catch (error) {
    console.error('Error while registering Nano Contract.', error);
    yield put(nanoContractRegisterError(t`Error while registering the nano contract.`));
    return;
  }

  if (!nanoUtils.isNanoContractCreate(tx)) {
    console.debug('Transaction is not a nano contract creation transaction.');
    yield put(nanoContractRegisterError(t`Invalid nano contract creation transaction.`));
    return;
  }
```
https://github.com/HathorNetwork/hathor-wallet/blob/master/src/sagas/nanoContract.js

What do you think?

Comment (alexruzenhack) [reply src/sagas/nanoContract.js]: Replaced: 4b81d31958257d4050a6af348bf85023fc5ed77c

File: locale/pt-br/texts.po [hunk 2]
```diff
@@ -57,7 +57,7 @@ msgid "[Last] dddd [•] HH:mm"
 msgstr "[Última] dddd [•] HH:mm"
 
 #.  See https://momentjs.com/docs/#/displaying/calendar-time/
-#: src/models.js:107 src/utils.js:426
+#: src/models.js:107 src/utils.js:427
```
Comment (andreabadesso) [inline locale/pt-br/texts.po:60]: Why this line changed in `src/utils`?

File: locale/texts.pot [hunk 3]
```diff
@@ -48,7 +48,7 @@ msgid "[Last] dddd [•] HH:mm"
 msgstr ""
 
 #: src/models.js:107
-#: src/utils.js:426
+#: src/utils.js:427
```
Comment (andreabadesso) [inline locale/texts.pot:51]: Why this changed? You didn't change `src/utils`

File: src/sagas/nanoContract.js [hunk 4]
```diff
@@ -45,7 +46,8 @@ export const failureMessage = {
   walletNotReadyError: t`Wallet is not ready yet to register a Nano Contract.`,
   addressNotMine: t`The informed address does not belong to the wallet.`,
   nanoContractStateNotFound: t`Nano Contract not found.`,
-  nanoContractStateFailure: t`Error while trying to get Nano Contract state.`,
+  nanoContractFailure: t`Error while trying register Nano Contract.`,
```
Comment (andreabadesso) [inline src/sagas/nanoContract.js:49]:
```suggestion
  nanoContractFailure: t`Error while trying to register Nano Contract.`,
```

**Review by andreabadesso — CHANGES_REQUESTED** [review]

**Review by pedroferreira1 — APPROVED** [review]

File: locale/pt-br/texts.po (hunk 2 above)
Comment (alexruzenhack) [reply locale/pt-br/texts.po:60]: I ran an update, probably it was lacking an update.

File: locale/texts.pot (hunk 3 above)
Comment (alexruzenhack) [reply locale/texts.pot:51]: I ran an update, probably it was lacking an update.

File: src/sagas/nanoContract.js (hunk 4 above)
Comment (alexruzenhack) [reply src/sagas/nanoContract.js]: Fixed: 003a3f8af07e1f5cb783427c7378624a816a9567

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**